### ISSUE TITLE
JSON-RPC API: Set methods have a huge flaw.

### ISSUE THREAD

**Issue by FRDS** [SoftEtherVPN/SoftEtherVPN#1117]
### Prerequisites

* [X] Can you reproduce?
* [ ] Are you running the latest version of SoftEtherVPN? 
Might not be, since I think there's a newer build.

#### VPN Server
**SoftEther version: 4.34 Build 9745 (2020/04/05)**
**Operating system: Linux Ubuntu 16.04.6 LTS**
**Architecture: 64-bit** 

### Description
After giving up using the JavaScript client library because of an [issue](https://github.com/SoftEtherVPN/SoftEtherVPN/issues/1088), I've build my external application to just use JSON requests to my SoftEther VPN Server.

But I found that Set methods in JSON-RPC API have a huge flaw, that it resets all the whole object's parameters if you don't include them in the request. This is bad if you're planning to remotely change a user's details like group assignation and/or policies, but currently you must include all the current user's credentials and settings like password. I personally don't choose to save my user's information and passwords to respect their privacy.

Please consider to modify the set method to not reset all the other parameters if it's not included in the request. This makes sense on create methods but not on set methods. I hope that this change will make server moderation easier and safer.

For example, if  I just wanted to "ban" a user by disabling their "Allow Access" policy:
```json
{
  "jsonrpc": "2.0",
  "id": "rpc_call_id",
  "method": "SetUser",
  "params": {
    "HubName_str": "hubname",
    "Name_str": "name"
    "UsePolicy_bool": true,
    "policy:Access_bool": false
  }
}
````

**Expected behavior:**
The object should only have its "Allow Access" policy disabled.

**Actual behavior:**
The user object had almost all their credentials and policies reset to default settings. Hence the user lost their credentials and settings for the VPN.

**Comment by davidebeatrici** [comment]
This is actually the intended behavior: a JSON client can get the object, modify it as desired and then set it.

The main reason for implementing it like this is due to the already existing RPC system (`PACK`), which sets the entire structure rather than single members.

**Comment by FRDS** [comment]
> This is actually the intended behavior: a JSON client can get the object, modify it as desired and then set it.
> 
> The main reason for implementing it like this is due to the already existing RPC system (`PACK`), which sets the entire structure rather than single members.

Do you know of an existing alternative that can let me modify just few details instead of the whole structure?

**Comment by davidebeatrici** [comment]
We only have those two RPC systems right now, however we can consider changing the behavior of the JSON one.

**Comment by FRDS** [comment]
> We only have those two RPC systems right now, however we can consider changing the behavior of the JSON one.

Thanks for the consideration. Banning users using IP Access Rules still have loopholes and policy settings have been the best option for us. It's just that doing it manually has been complicated, especially since we have _multiple servers_ to maintain.

**Comment by FRDS** [comment]
@davidebeatrici Hello, I'm just wondering if you guys still considering to change the JSON RPC behavior?

**Comment by davidebeatrici** [comment]
Hello, changing the behavior is fine. We aren't working on it yet, though.

**Comment by metalefty** [comment]
I agree that current behaviour is suck. I'll have a look when I have time.
